heidrun/ia.rb

```diff
@@ -0,0 +1,238 @@
+  end
+
+  # dpla:SourceResource
+  sourceResource class: DPLA::MAP::SourceResource do
+    # dcterms:isPartOf
+    #   meta.xml <sponsor>
+    collection class: DPLA::MAP::Collection,
+               each: record.field('sponsor'),
+               as: :collection do
+      title collection
+    end
+
+    # dcterms:creator
+    #   META.xml <creator>
+    creator class: DPLA::MAP::Agent,
+            each: record.field('creator'),
+            as: :creator do
+      providedLabel creator
+    end
+
+    # dc:date
+    #   META.xml <date>
+    date class: DPLA::MAP::TimeSpan,
+         each: record.field('date'),
+         as: :date do
+      providedLabel date
+    end
+
+    # dcterms:description
+    #   meta.xml <description>
+    description record.field('description')
+
+    # dcterms:extent
+    #   MARC.XML <datafield tag="300">[any subfields]
+    extent record.field('marc', 'record', 'datafield')
+      .match_attribute(:tag, '300')
+
+    # dcterms:identifier
+    #   META.XML
+    #     <identifier> may be the same thing;
+    #     <call_number>Call number: "[value]"</call_number>;
+    #     <datafield tag="035" ind1=" " ind2=" ">
+    #       <subfield code="a">[value]</subfield>
+    identifier record.map(&identifier_map).flatten
+
+    # dcterms:language
+    #   META.XML <language>
+    language record.field('language')
+
+    # dcterms:spatial
+    #   MARC.xml
+    #     <datafield tag="6##"><subfield code="z">
+    #     [ANY 600-699 data field
+    #      but only subfield "z" which is a geographic indicator]
+    spatial class: DPLA::MAP::Place,
+            each: record.field('marc', 'record', 'datafield')
+                .match_attribute(:tag) { |tag| tag.start_with?('6') }
+                .field('subfield')
+                .match_attribute(:code, 'z'),
+            as: :place do
+      providedLabel place
+    end
+
+    # dcterms:publisher
+    #   meta.xml <publisher>
+    publisher class: DPLA::MAP::Agent,
+              each: record.field('publisher'),
+              as: :publisher do
+      providedLabel publisher
+    end
+
+    # dc:relation
+    #   MARC.XML
+    #     <datafield tag="440"> or
+    #     <datafield tag="490"> or
+    #     <datafield tag="800"> or
+    #     <datafield tag="810"> or
+    #     <datafield tag="830">;
+    #     <datafield tag="785">
+    #       [Add ". " between subfields; Do not include subfield w];
+    #     <datafield tag="780">
+    #       [Add ". " between subfields; Do not include subfield w]
+    relation record.map(&relation_map).flatten
+
+    # dc:rights
+    #   "Access to the Internet Archive's Collections is granted
+    #    for scholarship and research purposes only. Some of the
+    #    content available through the Archive may be governed by
+    #    local, national, and/or international laws and regulations,
+    #    and your use of such content is solely at your own risk."
+    rights 'Access to the Internet Archive’s Collections is granted
```

Comment:
@guegueng In the old system, we're first pulling in the `<rights/>` and `<possible-copyright-status/>` elements from the `*_meta.xml`, and then using this text if there's nothing there. Do we want to still do that?